src/usage/context/usage.tsx

```diff
@@ -132,6 +165,75 @@ export const UsageProvider: FC<Props> = React.memo(({children}) => {
     handleGetBillingDate()
   }, [handleGetBillingDate])
 
+  const getComputedUsage = (vector_name: string, csvData: string) => {
+    // Calculation Formula
+    // Credit usage: $250 - (
+    //   (sum of 30-day writes * $0.002) +
+    //   (sum of 30-day query count * $0.01 / 100) +
+    //   (sum of 30-day query storage * $0.02) +
+    //   (sum of 30-day data out * $0.09)
+    //  )
+    const vectors = {
+      storage_gb: v => v * 0.02,
+      writes_mb: v => v * 0.002,
+      reads_gb: v => v * 0.09,
+      query_count: v => (v * 0.01) / 100,
+    }
+
+    const values = fromFlux(csvData).table.getColumn(vector_name) as Array<any>
+    if (!values) {
+      return 0
+    }
+
+    const tot = values.reduce((a, b) => a + b, 0)
+    return vectors[vector_name](tot)
+  }
+
+  const handleGetCreditUsage = useCallback(() => {
+    try {
+      setCreditUsage({
+        ...creditUsage,
+        status: RemoteDataState.Loading,
+      })
+      const vectors = ['storage_gb', 'writes_mb', 'reads_gb', 'query_count']
+      const promises = []
+
+      vectors.forEach(vector_name => {
+        promises.push(
+          getUsage({vector_name, query: {range: `${PAYG_CREDIT_DAYS}d`}}).then(
+            resp => {
+              if (resp.status !== 200) {
+                throw new Error(resp.data.message)
+              }
+
+              return new Promise(resolve =>
+                resolve(getComputedUsage(vector_name, resp.data))
+              )
+            }
+          )
+        )
+      })
+
+      Promise.all(promises).then(result => {
+        const amount: number = result
+          .reduce((a: number, b) => a + parseFloat(b), 0)
+          .toFixed(2)
+        setCreditUsage({
+          amount,
+          status: RemoteDataState.Done,
+        })
+      })
+    } catch (error) {
+      setCreditUsage({
+        ...creditUsage,
+        status: RemoteDataState.Done,
+      })
+    }
+  }, [])
```

Comment:
We may need  to either pass in the credit usage or list it as a dependency

Reply:
good catch!